pkg/controller/queuejobresources/genericresource/genericresource.go

```diff
@@ -175,8 +189,9 @@ func (gr *GenericResources) Cleanup(aw *arbv1.AppWrapper, awr *arbv1.AppWrapperG
 	// Check to see if object already exists in etcd, if not, create the object.
 	if inEtcd != nil || len(inEtcd.Items) > 0 {
 		newName := name
-		if len(newName) > 63 {
-			newName = newName[:63]
+		if len(newName) > 60 {
+			newName = newName[:60]
+			newName += GetRandomString(3)
 		}
 
 		err = deleteObject(namespaced, namespace, newName, rsrc, dclient)
```

Comment:
I think this will almost always fail for length >60 because the string you create is not deterministic. You should use a hash for getting the last 3 digits